File: software/examples/java/ExampleCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletHeartRate;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XYZ"; // Change to your UID

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletHeartRate hr = new BrickletHeartRate(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Set period for heart rate callback to 1s (1000ms)
		// Note: The heart rate callback is only called every second
		//       if the heart rate has changed since the last call!
		hr.setHeartRateCallbackPeriod(1000);

		// Add heart rate listener (parameter has unit bpm)
		hr.addHeartRateListener(new BrickletHeartRate.HeartRateListener() {
			public void heartRate(int heartRate) {
				System.out.println("Heart Rate: " + heartRate + " bpm");
			}
		});

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/go/example_callback.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/heart_rate_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Heart Rate Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	hr, _ := heart_rate_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	hr.RegisterHeartRateCallback(func(heartRate uint16) {
		fmt.Printf("Heart Rate: %d bpm\n", heartRate)
	})

	// Set period for heart rate receiver to 1s (1000ms).
	// Note: The heart rate callback is only called every second
	//       if the heart rate has changed since the last call!
	hr.SetHeartRateCallbackPeriod(1000)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
